client.go

```diff
@@ -123,10 +136,32 @@ func NewClient() *Client {
 	}
 }
 
+// ErrStopRetry is returned by a CheckRetry callback to stop retries, but
+// still have the client return the original error.
+var ErrStopRetry = errors.New("stop retry")
+
 // Do wraps calling an HTTP method with retries.
 func (c *Client) Do(req *Request) (*http.Response, error) {
 	c.Logger.Printf("[DEBUG] %s %s", req.Method, req.URL)
 
+	if c.CheckRetry == nil {
+		// Check the response code. We retry on 500-range responses to allow
+		// the server time to recover, as 500's are typically not permanent
+		// errors and may relate to outages on the server side.
+		c.CheckRetry = func(resp *http.Response, err error) error {
+			if err != nil {
+				return nil
+			}
+			// continue with our default policy of retrying server errors
+			// also checking for the occasional odd response codes.
+			if resp.StatusCode == 0 || resp.StatusCode >= 500 {
+				return nil
+			}
+
+			return ErrStopRetry
```

Comment:
We're only required to close the body if we return our own error when there was a valid response. Since this just signals to stop retrying, the caller can handle the `resp, error` combination normally.  (this will still apply with the new `(bool, error)` signature too)